### PR TITLE
Simplify clearing of spike buffers

### PR DESCRIPTION
**Pull request by mlober**
This PR fixes issue #2991 .

### CODE REVIEW

**Review by heplesser — APPROVED** [review]

File: nestkernel/event_delivery_manager.cpp [hunk 1]
```diff
@@ -116,17 +116,15 @@ void
 EventDeliveryManager::finalize()
 {
   // clear the spike buffers
-  for ( auto it = emitted_spikes_register_.begin(); it < emitted_spikes_register_.end(); ++it )
+  for ( auto vec_spikedata_ptr : emitted_spikes_register_ )
```
Comment (terhorstd) [inline nestkernel/event_delivery_manager.cpp:119]:
```suggestion
  for ( auto& vec_spikedata_ptr : emitted_spikes_register_ )
```

File: nestkernel/event_delivery_manager.cpp [hunk 2]
```diff
@@ -116,17 +116,15 @@ void
...
   {
-    ( *it )->clear();
-    delete ( *it );
+    delete vec_spikedata_ptr;
   }
-  emitted_spikes_register_.clear();
+  emitted_spikes_register_.clear(); // remove stale pointers
 
-  for ( auto it = off_grid_emitted_spikes_register_.begin(); it < off_grid_emitted_spikes_register_.end(); ++it )
+  for ( auto vec_spikedata_ptr : off_grid_emitted_spikes_register_ )
```
Comment (terhorstd) [inline nestkernel/event_delivery_manager.cpp:125]:
```suggestion
  for ( auto& vec_spikedata_ptr : off_grid_emitted_spikes_register_ )
```

**Review by terhorstd — CHANGES_REQUESTED** [review]
Minor changes necessary to stay consistent with other loops and not copy elements of the vector.

**Review by terhorstd — APPROVED** [review]
LGTM :+1:

**Pull request merged**